pkg/compactor/compactor.go

```diff
@@ -55,40 +55,38 @@ func (cp *Compactor) Compact(ctx context.Context, opts *brtypes.CompactOptions)
 	cp.logger.Info("Start compacting")
 
 	// Deepcopy restoration options ro to avoid any mutation of the passing object
-	cmpctOptions := opts.RestoreOptions.DeepCopy()
+	compactorRestoreOptions := opts.RestoreOptions.DeepCopy()
 
-	// If no basesnapshot is found, abort compaction as there would be nothing to compact
-	if cmpctOptions.BaseSnapshot == nil {
+	// If no base snapshot is found, abort compaction as there would be nothing to compact
+	if compactorRestoreOptions.BaseSnapshot == nil {
```

Comment:
Since this method `func (cp *Compactor) Compact()` can be called from anywhere (since it's publicly available), it's better to have this base snapshot check within this method as well. Anyway, I've simply changed the variable names in the compactor package, and no logic changes have been made.

Reply:
Sure, just that since it's publicly accessible it's better to keep it this way, atleast for now.

Reply:
Yes then we need a well defined API and define properly how to use backup-restore as an executable and how to use backup-restore API programmatically which means as a library. There is currently no use case for it to be used as a library and i think it was also never designed keeping that in mind.